File: Classes/Attacks/PlayerAttack.py
from Classes.Attacks.Attacks import Attack

class AtkPersonagem(Attack):
    def __init__(self, nomeAtk, danoAtk, spriteSheet, limiteLoops,
                 qtdFrames, pixels, spritePosition, offsetFrames, atkMana, atkButton):
        """
        :param nomeAtk: str
        :param danoAtk: int
        :param spriteSheet: spriteSheet loaded by pygame.image.load()
        :param limiteLoops: int
        :param pixels: (int, int) (pixels width, pixels height)
        :param qtdFrames: int
        :param spritePosition: (float, float)
        :param offsetFrames: float (dictates the animation speed)
        :param atkMana: int
        :param atkButton: ButtonWithText
        """

        Attack.__init__(self, nomeAtk, danoAtk, spriteSheet, limiteLoops,
                        qtdFrames, pixels, spritePosition, offsetFrames)

        self._atkMana = atkMana
        self._atkButton = atkButton

    def getAtkButton(self):
        return self._atkButton

    def setAtkButton(self, b):
        self._atkButton = b

    def getMana(self):
        return self._atkMana

    def setMana(self, c):
        self._atkMana = c

File: TheFlameRealmJava/core/src/main/java/com/flamerealm/game/attacks/PlayerAttack.java
package com.flamerealm.game.attacks;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.GridPoint2;
import com.badlogic.gdx.math.Vector2;
import com.flamerealm.game.ui.TextButton;

/**
 * Port de PlayerAttack.py (AtkPersonagem).
 */
public class PlayerAttack extends Attack {
    private int mana;
    private final TextButton atkButton;

    public PlayerAttack(String name, int damage, Texture spriteSheet, int limiteLoops,
                         int qtdFrames, GridPoint2 pixels, Vector2 spritePosition, float offsetFrames,
                         int mana, TextButton atkButton) {
        super(name, damage, spriteSheet, limiteLoops, qtdFrames, pixels, spritePosition, offsetFrames);
        this.mana = mana;
        this.atkButton = atkButton;
    }

    public TextButton getAtkButton() {
        return atkButton;
    }

    public int getMana() {
        return mana;
    }

    public void setMana(int mana) {
        this.mana = mana;
    }
}
